scripts/core/editor3/components/links/AttachmentList.tsx

```diff
@@ -22,38 +24,47 @@ export class AttachmentList extends React.Component<any, any> {
             });
     }
 
-    render() {
-        if (!this.state.attachments) {
-            return <div />; // wait for attachments
+    renderAttachment(attachment: IAttachment) {
```

Comment:
Extract this into an `AttachmentItem` component. I've noticed that when people put JSX in a method other than `render`, most of the time the code better of being a separate component. Ignore that no components in a single file rule. That only matters if there's a significant number of state in the file scope.